### PR TITLE
[TAP-577] Fixed an issue with S3/MinIO VFS file resolution.

### PR DESCRIPTION
**Pull request by jgminder**

### CODE REVIEW

File: legacy/src/main/java/org/pentaho/amazon/AmazonS3FileSystemBootstrap.java [hunk 1]
```diff
@@ -42,28 +46,37 @@
...
+      if ( fsm instanceof DefaultFileSystemManager && !Arrays.asList( fsm.getSchemes() )
+        .contains( S3FileProvider.SCHEME ) ) {
+        ( (DefaultFileSystemManager) fsm ).addProvider( S3FileProvider.SCHEME, new S3FileProvider() );
       }
+
+      if ( connectionManager.get() != null ) {
+        connectionManager.get().addConnectionProvider( S3FileProvider.SCHEME, new S3Provider() );
+      }
+
+      VFSLookupFilter vfsLookupFilter = new VFSLookupFilter();
```
Comment (dmoore62) [inline legacy/src/main/java/org/pentaho/amazon/AmazonS3FileSystemBootstrap.java:75]: I feel like all of this should be in the null guard on line 71

File: legacy/src/main/java/org/pentaho/amazon/AmazonS3NFileSystemBootstrap.java [hunk 2]
```diff
@@ -46,32 +47,37 @@
...
+        .contains( S3NFileProvider.SCHEME ) ) {
+        ( (DefaultFileSystemManager) fsm ).addProvider( S3NFileProvider.SCHEME, new S3NFileProvider() );
       }
+
       if ( connectionManager.get() != null ) {
         connectionManager.get().addConnectionProvider( S3FileProvider.SCHEME, new S3Provider() );
       }
+
+      VFSLookupFilter vfsLookupFilter = new VFSLookupFilter();
+      connectionManager.get().addLookupFilter( vfsLookupFilter );
```
Comment (dmoore62) [inline legacy/src/main/java/org/pentaho/amazon/AmazonS3NFileSystemBootstrap.java:77]: Same down here too

File: legacy/src/main/java/org/pentaho/amazon/AmazonS3FileSystemBootstrap.java (hunk 1 above)
Comment (jgminder) [reply legacy/src/main/java/org/pentaho/amazon/AmazonS3FileSystemBootstrap.java:75]: Cut and pasted. I agree, let me update.

File: legacy/src/main/java/org/pentaho/amazon/AmazonS3NFileSystemBootstrap.java (hunk 2 above)
Comment (jgminder) [reply legacy/src/main/java/org/pentaho/amazon/AmazonS3NFileSystemBootstrap.java:77]: Cut and pasted. I agree, let me update.

File: legacy/src/main/java/org/pentaho/amazon/AmazonS3FileSystemBootstrap.java (hunk 1 above)
Comment (baudekin) [reply legacy/src/main/java/org/pentaho/amazon/AmazonS3FileSystemBootstrap.java:75]: I agree I don't think we want to get the lookup filter if the connectionManager is null.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 42m 6s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl legacy
```
---
:ok_hand: All tests passed!
 
Tests run: **549**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-plugin/job/PR-2080/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 4m
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl legacy
```
---
:ok_hand: All tests passed!
 
Tests run: **549**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/big-data-plugin/job/PR-2080/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**